### DISCUSSION TITLE
Пересобрать многомерный массив в одномерный

### DISCUSSION THREAD

**Discussion by ToeraHL** [totumonline/totum-ru-issues-and-discussions#284] [Как это сделать?] [answered]
Здравствуйте. Подскажите, пожалуйста, в какую сторону думать.
Есть многомерный массив вида:
 [ 
    {
        a = 12,
        b = [1,2,3]
    },
    {
    a = 25, 
    b = [8,2]
    },
] 

нужно его привести к виду:
 [ 
    {
        a = 12,
        b = 1
    },
    {
        a = 12,
        b = 2
    },
    {
        a = 12, 
        b = 3
    },
....
] 

На ум приходит два цикла друг в друге, но тогда нужна какая-то переменная, которая бы копила значения массива, а не перезаписывала. Или логика должна быть совсем другая?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
```
= : listTrain(list: $replace)

replace: listReplace(list: #h_data; action: $new_val; value: "value")

	new_val: rowListCreate(field: "a" = $#value[a]; field: "b" = $#value[b])
```

В `h_data` исходные данные. Результат:

```
[
  {
    "a": "12",
    "b": "1"
  },
  {
    "a": "12",
    "b": "2"
  },
  {
    "a": "12",
    "b": "3"
  },
  {
    "a": "25",
    "b": "8"
  },
  {
    "a": "25",
    "b": "2"
  }
]
```

**Reply by ttm-alexeypolunin** [reply]
Можно и через while цикл. Переменную может хранить функция var. Но для Тотум циклы в while это не круто.

**Reply by ToeraHL** [reply]
Тогда действительно лучше первый способ.  Получается, var поддерживает конкатенацию?

**Reply by ttm-alexeypolunin** [reply]
Не понял вопрос. var записывает переменную целиком. Если нужно в нее последовательно что-то добавлять то переменная вызывается, делается добавление (конкатенация) и записывается обратно.

**Reply by ttm-alexeypolunin** [reply]
Если хочется с var какими-нибудь заморочится, то вот прям топчик 80 lvl: https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/58

**Reply by ToeraHL** [reply]
А, понятно, спасибо
